tests/unit/backend/recipe/algorithm/test_FocusSpectraAlgorithm.py

```diff
@@ -0,0 +1,259 @@
+)
+from snapred.meta.Config import Resource
+
+
+class TestFocusSpectra(unittest.TestCase):
+    def setUp(self):
+        """Create a set of mocked ingredients for calculating DIFC corrected by offsets"""
+        self.maxOffset = 2
+        self.fakeRunNumber = "555"
+
+        self.fakeInstrumentState = InstrumentState.parse_raw(Resource.read("/inputs/diffcal/fakeInstrumentState.json"))
+
+        self.fakeIngredients = Resource.read("/inputs/reduction/input_ingredients.json")
+        self.fakeIngredients = ReductionIngredients.parse_raw(self.fakeIngredients)
+        self.fakeIngredients.pixelGroup = self.fakeInstrumentState.pixelGroup
+
+        self.fakeRawData = f"_test_focusSpectra_{self.fakeRunNumber}"
+        self.fakeGroupingWorkspace = f"_test_focusSpectra_difc_{self.fakeRunNumber}"
+        self.makeFakeNeutronData(self.fakeRawData, self.fakeGroupingWorkspace)
+
+    def tearDown(self) -> None:
+        for ws in mtd.getObjectNames():
+            try:
+                DeleteWorkspace(ws)
+            except:  # noqa: E722
+                pass
+        return super().tearDown()
+
+    def makeFakeNeutronData(self, rawWsName: str, focusWSname: str) -> None:
+        """Create sample data for test runs"""
+        from mantid.simpleapi import (
+            ConvertUnits,
+            CreateSampleWorkspace,
+            LoadDetectorsGroupingFile,
+            LoadInstrument,
+            Rebin,
+            RebinRagged,
+        )
+
+        TOFMin = self.fakeInstrumentState.particleBounds.tof.minimum
+        TOFMax = self.fakeInstrumentState.particleBounds.tof.maximum
+
+        dMin = self.fakeIngredients.pixelGroup.dMin()
+        dMax = self.fakeIngredients.pixelGroup.dMax()
+        DBin = self.fakeIngredients.pixelGroup.dBin()
+        overallDMax = max(dMax)
+        overallDMin = min(dMin)
+        dBin = min([abs(d) for d in DBin])
+
+        # prepare with test data
+        midpoint = (overallDMin + overallDMax) / 2.0
+        CreateSampleWorkspace(
+            OutputWorkspace=rawWsName,
+            # WorkspaceType="Histogram",
+            Function="User Defined",
+            UserDefinedFunction=f"name=Gaussian,Height=10,PeakCentre={midpoint},Sigma={midpoint/10}",
+            Xmin=overallDMin,
+            Xmax=overallDMax,
+            BinWidth=abs(dBin),
+            XUnit="dSpacing",
+            NumBanks=4,  # must produce same number of pixels as fake instrument
+            BankPixelWidth=2,  # each bank has 4 pixels, 4 banks, 16 total
+        )
+        LoadInstrument(
+            Workspace=rawWsName,
+            Filename=Resource.getPath("inputs/testInstrument/fakeSNAP.xml"),
+            RewriteSpectraMap=True,
+        )
+        # also load the focus grouping workspace
+        LoadDetectorsGroupingFile(
+            InputFile=Resource.getPath("inputs/testInstrument/fakeSNAPFocGroup_Natural.xml"),
+            InputWorkspace=rawWsName,
+            OutputWorkspace=focusWSname,
+        )
```

Comment:
It should be possible to use the new synthetic data here

Reply:
I dont think that's in next yet.